pkg/util/namespacedname.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+// same as k8s.io/apimachinery/pkg/types/namespacedname.go, but with JSON (de)serialization
+
+import (
+	"fmt"
+)
+
+// FIXME: K8s uses '/' as its separator. We currently use this for backwards-compatibility, but we
+// should change to '/'. That change will involve fixing the many places where we create strings
+// with %s:%s instead of %s/%s as well; those should just use NamespacedName directly.
+const Separator = ':'
+
+// NamespacedName represents a resource name with the namespace it's in.
+//
+// When printed with '%s' or '%v', NamespacedName is rendered as "<namespace>:<name>". Printing with
+// '%+v' or '%#v' renders as it would normally.
+type NamespacedName struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+func (n NamespacedName) Format(state fmt.State, verb rune) {
+	if verb == 'v' {
```

Comment:
I've always seen these format methods implemented with switch statements, which I think is a little clearer to read, but thias preference.